### DISCUSSION TITLE
Help me understand purpose of Immich video transcoding (and how to properly use it).

### DISCUSSION THREAD

**Discussion by erkexzcx** [immich-app/immich#14030] [Q&A] [answered]
Hello. This question might sound a completely dumb, but I am struggling to understand the point of video transcoding, or more precisely - what is recommended?

Previously I thought that original videos are replaced with encoded ones - using AV1 to save storage is great, but later on I realized that it's just for better playback compatibility:

![image](https://github.com/user-attachments/assets/986b93c1-ce45-4c8f-bc66-c866280f64cc)

![image](https://github.com/user-attachments/assets/34967b1a-d890-49f2-b6e9-1511b09e114c)

So I assume the goal is to transcode videos for better playback compatibility in apps and browser, right? Then I have few questions:

1. All my videos are made by using phone, which means all formats are compatible with browser and Android app. I don't need transcoding at all?
2. Let's say some format is not supported, so I would need transcoding - can anyone suggest what kind of settings should I use? It seems my Intel NUC doesn't support AV1 codec, while the rest is supported (hardware accel). So I should accept everything, but AV1 and transcode it to, let's say, VP9? Any generic tutorial on Immich transcoding settings?
3. Does Immich have any plans to replace original files with transcoded ones, so disk space can be saved?

**Comment by bo0tzz** [comment, marked as answer]
1. You can turn transcoding off if you want, but compatability isn't guaranteed
2. The defaults are fine
3. No

**Comment by Sawtaytoes** [comment]
I was wondering about this myself.

When viewing videos, if the original is already compatible, is it going to use the transcoded one or will it play the original file?
